core/src/main/java/org/apache/brooklyn/enricher/stock/Propagator.java

```diff
@@ -195,7 +196,15 @@ public void emitAllAttributes(boolean includeNullValues) {
         }
     }
 
-    private Sensor<?> getDestinationSensor(Sensor<?> sourceSensor) {
-        return sensorMapping.containsKey(sourceSensor) ? sensorMapping.get(sourceSensor): sourceSensor;
+    private Sensor<?> getDestinationSensor(final Sensor<?> sourceSensor) {
+        Optional<? extends Sensor<?>> mappingSensor = Iterables.tryFind(sensorMapping.keySet(), new Predicate<Sensor<?>>() {
```

Comment:
Worth a comment saying why we're doing this, rather than the old simpler `sensorMapping.get(sourceSensor)`. e.g.:

    // sensor equality includes the type; we want just name-equality so will use predicate.